Source/MQTTnet/MqttApplicationMessage.cs

```diff
@@ -50,9 +51,9 @@ public sealed class MqttApplicationMessage
         public uint MessageExpiryInterval { get; set; }
 
         /// <summary>
-        /// Get or set ArraySegment style of Payload.
+        /// Get or set ReadOnlySequence style of Payload.
         /// </summary>
-        public ArraySegment<byte> PayloadSegment { get; set; } = EmptyBuffer.ArraySegment;
+        public ReadOnlySequence<byte> PayloadSequence { get; set; }
```

Comment:
In my opinion we should rename the property to "Payload" because the type of suffix was introduced because they were multiple types of payloads in the same class.

Reply:
renamed most what was PayloadSequence to Payload, also some temp variables